layouts/partials/helpers/fragments.html

```diff
@@ -76,8 +76,13 @@
 {{- $root := $layout_info.root -}}
 {{- $is_404 := $layout_info.is_404 -}}
 
+{{ $default_lang := $real_page.Site.Params.DefaultContentLanguage }}
+{{ $default_site := index (where $real_page.Sites "Language.Lang" "eq" $default_lang) 0 }}
+{{ $lang := $real_page.Site.Language.Lang }}
+
 {{- $page_scratch.Set "fragment_directories" (slice) -}}
 {{- if eq $is_404 true -}}
+  {{- $page_scratch.Add "fragment_directories" ($default_site.GetPage "page" "/_global") -}}
   {{- $page_scratch.Add "fragment_directories" ($real_page.Site.GetPage "page" "/_global") -}}
```

Comment:
Wouldn't that duplicate instead of replace fragments?

Reply:
`default_site` is different that `real_page.Site` if the page is being generated for something other than default language. In case page language is default, the code responsible for removing duplicate fragments takes care it. Here we just find what directories may contain fragments, and later read fragment from them and after that remove overriding fragments.

Reply:
Thanks for the context.